### DISCUSSION TITLE
Custom response for `reset_password_router` and `verify_router`

### DISCUSSION THREAD

**Discussion by Abdeldjalil-H** [fastapi-users/fastapi-users#1472] [Q&A]
I want to return custom response in `/forgot-password` endpoint. For example instead of returning `None` I want to return 
```python
{
    "message": "If the email is registered, you will receive instructions to reset your password"
}
```
the `BaseUserManager.on_after_forgot_password` method does not take `response` object like `on_after_login`. I want to know if there is a simple way to return a custom response? And why there is no response object for `on_after_forgot_password` and other ones related to reset password and verify user.

**Comment by hgalytoby** [comment]
You can read the instructions here, it's a security issue, as is `/request-verify-token`.

https://fastapi-users.github.io/fastapi-users/latest/usage/routes/#post-forgot-password

To prevent malicious users from guessing existing users in your database, the route will always return a 202 Accepted response, even if the user requested does not exist.

**Reply by Abdeldjalil-H** [reply]
Thank you @hgalytoby for your response. I've already read this but I think there is no issue returning a message like the above one.

**Reply by hgalytoby** [reply]
You can use a middleware to solve this problem.

reference

https://github.com/fastapi-users/fastapi-users/discussions/1470#discussioncomment-11319523

**Reply by Abdeldjalil-H** [reply]
This works, but it will be much simpler if we have access to the `response` object. Also this will unify the signature of all methods.

**Reply by hgalytoby** [reply]
It seems to work, you can push an RP to try it out.

Otherwise, you can only solve the problem from middleware.
